sdk/core/Azure.Core/tests/RetryPolicyTestBase.cs

```diff
@@ -673,36 +673,36 @@ public RetryPolicyMock(RetryMode mode, int maxRetries = 3, TimeSpan delay = defa
 
             public AsyncGate<TimeSpan, object> DelayGate { get; } = new AsyncGate<TimeSpan, object>();
 
-            internal override void Wait(TimeSpan time, CancellationToken cancellationToken)
-            {
-                DelayGate.WaitForRelease(time).GetAwaiter().GetResult();
-            }
+            //internal override void Wait(TimeSpan time, CancellationToken cancellationToken)
```

Comment:
Given this need, would it make sense to create a RequestDelay object that implements wait and can be overridden with a gate?

Reply:
Added MessageDelay to address this.